tests/Tests.hs

```diff
@@ -105,9 +102,10 @@ instance Arbitrary Month where
 instance Arbitrary DateTime where
     arbitrary = DateTime <$> arbitrary <*> arbitrary
 instance Arbitrary Date where
-    arbitrary = Date <$> choose dateRange
-                     <*> arbitrary
-                     <*> choose (1,28)
+    arbitrary = do
+        year <- choose dateRange
+        month <- arbitrary
+        Date year month <$> choose (1, daysInMonth year month)
```

Comment:
This introduces a data dependency of the day on the other two fields, so you can't just apply the constructor to all three like before. I did't see an obvious way of doing this with Applicative that was reasonably clear, did I miss something obvious?